src/snake/core/sampling/samplers.py

```diff
@@ -219,6 +221,88 @@ def _single_frame(self, sim_conf: SimConfig) -> NDArray:
+    ----------
+    acsz: float | int
+        Number/ proportion of lines to be acquired in the center of k-space.
+    accelz: int
+        Acceleration factor for the rest of the lines.
+    directionz: Literal["center-out", "random"]
+        Direction of the acquisition. Either "center-out" or "random".
+    pdfz: Literal["gaussian", "uniform"]
+        Probability density function of the sampling. Either "gaussian" or "uniform".
+    obs_ms: int
+        Time spent to acquire a single shot
+    nb_revolutions: int
+        Number of revolutions of the spiral.
+    in_out: bool
+        If true, the spiral is acquired with a double join pattern from/to the periphery
+    **kwargs:
+        Extra arguments (smaps, n_jobs, backend etc...)
+    """
+
+    __sampler_name__ = "rotated-stack-of-spiral"
+    acsz: float | int
+    accelz: int
+    orderz: VDSorder = VDSorder.TOP_DOWN
+    nb_revolutions: int = 10
+    spiral_name: str = "archimedes"
+    pdfz: VDSpdf = VDSpdf.GAUSSIAN
+    constant: bool = False
+    in_out: bool = True
+    rotate_angle: AngleRotation = AngleRotation.ZERO
+    rotate_frame_angle: AngleRotation | int = 0
+    obs_time_ms: int = 30
+    n_shot_slices: int = 1
+    frame_index: int = 0
+
+    def fix_angle_rotation(
+        self, frame: Generator[np.ndarray, None, None], angle: AngleRotation | float = 0
+    ) -> Generator[np.ndarray, None, None]:
+        """Rotate the trajectory by a given angle."""
+        for traj in frame:
+            yield from rotate_trajectory((x for x in [traj]), angle)
+
+    def get_next_frame(self, sim_conf: SimConfig) -> NDArray:
+        """Generate the next rotated frame."""
+        base_frame = self._single_frame(sim_conf)
+        if self.constant or self.rotate_frame_angle == 0:
+            return base_frame
+        else:
+            self.frame_index += 1
+            rotate_frame_angle = np.pi * (self.rotate_frame_angle / 180)
+            base_frame_gen = (traj[None, ...] for traj in base_frame)
+            rotated_frame = self.fix_angle_rotation(
+                base_frame_gen, float(rotate_frame_angle * self.frame_index)
+            )
+            return np.concatenate(
+                [traj.astype(np.float32) for traj in rotated_frame], axis=0
+            )
+
+    def _single_frame(self, sim_conf: SimConfig) -> NDArray:
+        """Generate the sampling pattern."""
+        n_samples = int(self.obs_time_ms / sim_conf.hardware.dwell_time_ms)
+        single_frame = stack_spiral_factory(
+            shape=sim_conf.shape,
+            accelz=self.accelz,
+            acsz=self.acsz,
+            n_samples=n_samples,
+            nb_revolutions=self.nb_revolutions,
+            pdfz=self.pdfz,
+            orderz=self.orderz,
+            spiral=self.spiral_name,
+            rotate_angle=self.rotate_angle,
+            in_out=self.in_out,
+            n_shot_slices=self.n_shot_slices,
+            rng=sim_conf.rng,
+        )
+        return single_frame
```

Comment:
This is the same as the one from `StackOfSpiral` so you can inherit from it instead and avoid the repetition